### PR TITLE
Feature Converter doesn't cleanup after itself

### PR DESCRIPTION
**Pull request by kaushalmall**
added cleanup method to cleanup directories that the converter create. 

FYI @DominikSuess

### CODE REVIEW

File: src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java [hunk 1]
```diff
@@ -76,4 +82,15 @@ public void run() {
...
+    private void cleanUp(){
+        String tmpDir = System.getProperty("java.io.tmpdir");
+        logger.info("Cleaning up tmp directories {}, {}", tmpDir + "sub-content-packages", tmpDir + "syntethic-content-packages" );
+
+        try {
+            FileUtils.deleteDirectory( new File (tmpDir, "syntethic-content-packages") );
+            FileUtils.deleteDirectory( new File(tmpDir, "sub-content-packages") );
+        } catch (IOException e) {
+            logger.error( "Error Deleting {}, {}", tmpDir + "sub-content-packages", tmpDir + "syntethic-content-packages");
+        }
```
Comment (bosschaert) [inline src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java:94]: I would make `"sub-content-packages"` and `"syntethic-content-packages"` constants...

**Comment by DominikSuess** [comment]
Despite of the suggestions of @bosschaert  I think this PR is improving the situation - we could think of improving it to properly clean up after itself as well but for the issue at hand cleaning the working directories sounds like a good plan.

File: src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java [hunk 2]
```diff
@@ -76,4 +82,15 @@ public void run() {
         logger.info("+-----------------------------------------------------+");
     }
 
+    private void cleanUp(){
+        String tmpDir = System.getProperty("java.io.tmpdir");
+        logger.info("Cleaning up tmp directories {}, {}", tmpDir + "sub-content-packages", tmpDir + "syntethic-content-packages" );
```
Comment (DominikSuess) [inline src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java:87]: @kaushalmall - just a minor thing - tmpdir may not have separator at the end so the error message may actually not really return the right path, creating the the Dir fileobjects ahead and using the path out of those would be cleaner.

File: src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java (hunk 1 above)
Comment (kaushalmall) [reply src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java:94]: @bosschaert this was conflicting for me, I _was_ going to make them constants, but, the existing classes don't have them as constants, and I decided to follow the same practice for consistency sake. Maybe another issue to track making a common constants class across the board?

Comment (bosschaert) [reply src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java:94]: Ok fair enough. No problem then :)

File: src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java (hunk 2 above)
Comment (kaushalmall) [reply src/main/java/org/apache/sling/feature/cpconverter/cli/ShutDownHook.java]: @DominikSuess hopefully the latest change is better aligned to what you were thinking

**Comment by cziegeler** [comment]
I assume the problem here is a re-run of the tool?
If that is the case, why not delete the directory *before* running the tool; this way someone just running the tool once does not pay the price of waiting for the directory delete

It would also be good to have a jira issue for this

**Comment by bdelacretaz** [comment]
FWIW, I think "synthetic" is spelled wrong in `syntethic-content-packages` - but maybe that's a reflection of a typo made elsewhere.

**Comment by kaushalmall** [comment]
>why not delete the directory before running the tool; 
Hi @cziegeler @DominikSuess if this is preferred, I can move this code to the corresponding class(es) to delete _before_ the processing starts. Let me know.

**Comment by kaushalmall** [comment]
@bdelacretaz thanks for catching it, the original class has the same typo, but, I just updated the PR to fix it across the board.

**Comment by bosschaert** [comment]
> do you mind creating an issue for this? Not sure if I have access

Creating a sling issue can be done by anyone. Just create an account and file the issue here: https://issues.apache.org/jira/projects/SLING

**Comment by kaushalmall** [comment]
created https://issues.apache.org/jira/browse/SLING-9490
thanks @bosschaert

**Comment by kaushalmall** [comment]
hi @DominikSuess @cziegeler @bdelacretaz @bosschaert any reason for this to not be merged?

**Comment by cziegeler** [comment]
@kaushalmall Sorry, it's applied now - thanks!

**Pull request merged**